spec/acceptance/kubernetes_spec.rb

```diff
@@ -48,17 +49,17 @@ class {'kubernetes':
       end
 
       context 'application deployment' do
-
+        before(:all) { change_target_host('controller') }
+        after(:all) { reset_target_host }
         it 'can deploy an application into a namespace and expose it' do
-          run_shell('sleep 180')
           run_shell('KUBECONFIG=/etc/kubernetes/admin.conf kubectl create -f /tmp/nginx.yml') do |r|
             expect(r.stdout).to match(/my-nginx created\nservice\/my-nginx created\n/)
           end
         end
 
         it 'can access the deployed service' do
-          run_shell('sleep 60')
-          run_shell('curl -s 10.96.188.5') do |r|
+          sleep(180)
+          run_shell('curl -s 10.96.188.5', expect_failures: true) do |r|
```

Comment:
if you have a couple of failures with this call and you might want to add a retry mechanism here. see the --retry options for curl